Tests/xComputermanagement.Tests.ps1

```diff
@@ -0,0 +1,159 @@
+                Assert-MockCalled -CommandName Add-Computer -Exactly 1 -Scope It -ParameterFilter {$WorkGroupName -and $NewName}
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It -ParameterFilter {$DomainName}
+            }
+            It 'Changes only the Domain to new Domain' {
+                Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso.com';Workgroup='Contoso.com';PartOfDomain=$true}}
+                Mock GetComputerDomain {'contoso.com'}
+                Set-TargetResource -Name $Env:ComputerName -DomainName 'adventure-works.com' -Credential $Credential -UnjoinCredential $Credential | Should BeNullOrEmpty
+                Assert-MockCalled -CommandName Rename-Computer -Exactly 0 -Scope It
+                Assert-MockCalled -CommandName Add-Computer -Exactly 1 -Scope It -ParameterFilter {$DomainName}
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It -ParameterFilter {$NewName}
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It -ParameterFilter {$WorkGroupName}
+            }
+            It 'Changes only Domain to Workgroup' {
+                Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso.com';Workgroup='Contoso.com';PartOfDomain=$true}}
+                Mock GetComputerDomain {''}
+                Set-TargetResource -Name $Env:ComputerName -WorkGroupName 'Contoso' -UnjoinCredential $Credential | Should BeNullOrEmpty
+                Assert-MockCalled -CommandName Rename-Computer -Exactly 0 -Scope It
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It -ParameterFilter {$NewName}
+                Assert-MockCalled -CommandName Add-Computer -Exactly 1 -Scope It -ParameterFilter {$WorkGroupName}
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It -ParameterFilter {$DomainName}
+            }
+            It 'Changes only ComputerName in Domain' {
+                Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso.com';Workgroup='Contoso.com';PartOfDomain=$true}}
+                Mock GetComputerDomain {'contoso.com'}
+                Set-TargetResource -Name $NotComputerName -Credential $Credential | Should BeNullOrEmpty
+                Assert-MockCalled -CommandName Rename-Computer -Exactly 1 -Scope It
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It
+            }
+            It 'Changes only ComputerName in Workgroup' {
+                Mock GetComputerDomain {''}
+                Mock Get-WMIObject {[PSCustomObject]@{Domain = 'Contoso';Workgroup='Contoso';PartOfDomain=$false}}
+                Set-TargetResource -Name $NotComputerName | Should BeNullOrEmpty
+                Assert-MockCalled -CommandName Rename-Computer -Exactly 1 -Scope It
+                Assert-MockCalled -CommandName Add-Computer -Exactly 0 -Scope It
+            }
+        }
+    }
+}
```

Comment:
end file with new line per item 5 of the [Style Guidelines] (https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) refered to in the contribution guidelines.